### PR TITLE
MB-16978 EDI LOA for Coast Guard SMs should include 'HS' household goods code

### PR DESCRIPTION
**Pull request by PDickeyTruss**
## Summary

This PR adds an alternate query to fetch a Line of Accounting (LOA) for a given Transportation Accounting Code (TAC) for Service Members that are in the Coast Guard. The vast majority of the code in this PR is new unit tests and additional EDI test harness data. 

The business logic that was updated is a bit convoluted in the github diff. There are now two queries for getting records from the LOA table: a new query for Coast Guard service members, and the existing query for all other service members. The queries are very similar with the difference being the Coast Guard query does not do any selection logic based on service member rank and includes LOAs with a household goods code (loa_hs_gds_cd) of 'HS'. The existing (non-Coast Guard) query excludes household goods code 'HS'.  The below diff shows the existing non-Coast Guard query on the left and the Coast Guard only query on the right:

Screenshot 2023-09-11 at 11 29 03 AM

## Test harness data addition

Testing the changes made in this PR proved to be quite time consuming. To speed things up, I added a new test harness function to create a Coast Guard move with a payment request ready for EDI generation. This can be found under the name `MakeMoveWithReviewedServiceItemsReadyForEDI`. 

## PR Validation (Test data)
The easiest way to validate this PR is using the new test harness data:
- [ ] Go to the test harness data: `http://officelocal:3000/testharness/list`
- [ ] Click `MakeMoveWithReviewedServiceItemsReadyForEDI`
- [ ] Search for and retain the payment request number
- [ ] Generate an EDI locally for the retained payment request number `generate-payment-request-edi --payment-request-number <payment_request_number>`
- [ ] Inspect the FA2 lines of the EDI. There should be (test) data in ~25-30 different FA2 fields, for each of the five service items in the test payment request. 

https://github.com/transcom/mymove/assets/62263329/29111cf5-d498-4d3c-b3c8-e4ad9b484d09

## PR Validation ('Real' data)
This PR can (and has been) tested against real, non-test data as well. It is pretty time consuming and perhaps beyond the scope of what is necessary for PR review. Regardless, here are the steps I took to test this PR locally with staging data for the TAC code `ZNRC`:

* Get stg data (TACs/LOAs)
	* Run stg migrations locally `make run_stg_migrations`
	* Point server to `deployed_migrations` local db. Add to envrc.local: 
		`export DB_PORT=5434`
		`export DB_NAME=deployed_migrations`
	* [Related Docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#downloading-secure-migrations)
* Start server `make server_run` and client `make client_run`
* In the Customer app, create a new Coast Guard user/move all the way through move submission
* As a Services Counselor in the office app, find the created move, fill in remarks and TAC code `ZNRC`, submit move details
* As TOO, approve and send the move task order
* As a prime user, update shipment (dates and weights) and create a payment request (retain payment request #)
* As a TIO, go to payment requests of the move to review weights and review/approve service items. 
* Generate a EDI for the payment request `generate-payment-request-edi --payment-request-number 9430-9719-1`
    * [Related Docs](https://transcom.github.io/mymove-docs/docs/backend/testing/acceptance-testing-syncada-edi-invoicing#generate-edi-locally) 
* Look at the FA2 lines of the EDI, verify the data is from the "correct" LOA

### CODE REVIEW

**Review by mboren — APPROVED** [review]
Code looks good. I tested with the test data (not the real stuff) and that looks fine as well.

File: pkg/testdatagen/testharness/dispatch.go [hunk 1]
```diff
@@ -191,6 +191,9 @@ var actionDispatcher = map[string]actionFunc{
 	"MakeHHGMoveWithAddressChangeRequestAndSecondDeliveryLocation": func(appCtx appcontext.AppContext) testHarnessResponse {
 		return MakeHHGMoveWithAddressChangeRequestAndSecondDeliveryLocation(appCtx)
 	},
+	"MakeMoveWithReviewedServiceItemsReadyForEDI": func(appCtx appcontext.AppContext) testHarnessResponse {
```
Comment (mandaem) [inline pkg/testdatagen/testharness/dispatch.go:194]: This is great work! Do you think this scenario will be the best way to test the upcoming EDI work we have on deck?

Comment (mandaem) [reply pkg/testdatagen/testharness/dispatch.go:194]: I think I might have found out the answer after testing this work further.... the FA2*L1s I get are all six digits already, so we can't test 17008 this way without editting the scenario, particularly line 75 in pkg/factory/line_of_accounting_factory.go. Is that the best course of action in your opinion @PDickeyTruss ? For people who do any upcoming work to modify the scenario?

**Review by mandaem — APPROVED** [review]
I tried a couple and everything functions as expected.
Screenshot 2023-09-13 at 2 33 07 PM

File: pkg/testdatagen/testharness/dispatch.go (hunk 1 above)
Comment (PDickeyTruss) [reply pkg/testdatagen/testharness/dispatch.go:194]: You are on to it. 💯 Worth noting, changing the value on line 75 of the loa factory would change it for everywhere we use `BuildFullTransportationAccountingCode` (not a big deal in this case). This value could also be changed via customization when using the build function. 

I am gonna pull this testharness data creation into its own PR. Let's make sure we get the data updates you would like included in that PR.

**Comment by duncan-truss** [comment]
@PDickeyTruss I removed the Do Not Merge label, we just need to expand the requirement slightly to account for 'HT' and 'HC' household goods codes as well and then we can call this done.

**Review by kctruss — APPROVED** [review]
Approving as DB codeowner - looks like a migration filename was reordered in the manifest file? Not a big deal.

**Comment by PDickeyTruss** [comment]
@duncan-truss I worked out all the gross merge conflicts on this. There are now 2 EDI test data moves, the existing `MakeMoveReadyForEDI` from a previous PR and this PR adds `MakeCoastGuardMoveReadyForEDI`. 

I see there are still some ongoing discussions related to the query in this PR. As it sits this PR still filters by 'HS', should be easy enough to tweak once requirements are settled. ☺️

**Review by kctruss — APPROVED** [review]
Reviewed the additional changes from @duncan-truss, adding an additional approval.

**Pull request merged**